tests/backup/backup_test.go

```diff
@@ -8177,3 +8195,47 @@ func removeIntItemFromSlice(itemList []int, item []int) []int {
 	}
 	return itemList
 }
+
+func getAllBackupLocations(ctx context.Context) (map[string]string, error) {
+	backupLocationMap := make(map[string]string, 0)
+	backupDriver := Inst().Backup
+	backupLocationEnumerateRequest := &api.BackupLocationEnumerateRequest{
+		OrgId: orgID,
+	}
+	response, err := backupDriver.EnumerateBackupLocation(ctx, backupLocationEnumerateRequest)
+	if err != nil {
+		return nil, nil
+	}
+	if len(response.BackupLocations) > 0 {
+		log.Infof("Found backup locations - ")
+		for _, backupLocation := range response.BackupLocations {
+			backupLocationMap[backupLocation.Uid] = backupLocation.Name
+			log.Infof(backupLocation.Name)
+		}
+	} else {
+		log.Info("No backup locations found")
+	}
+	return backupLocationMap, nil
+}
+
+func getAllCloudCredentials(ctx context.Context) (map[string]string, error) {
+	cloudCredentialMap := make(map[string]string, 0)
+	backupDriver := Inst().Backup
+	cloudCredentialEnumerateRequest := &api.CloudCredentialEnumerateRequest{
+		OrgId: orgID,
+	}
+	response, err := backupDriver.EnumerateCloudCredential(ctx, cloudCredentialEnumerateRequest)
+	if err != nil {
+		return nil, nil
+	}
```

Comment:
Is there a reason you are not returning error here? Can we at least have some sort of message here like no cloud credentials found or something, if the error means that there were no credentials found in the org

Reply:
Good catch. Not sure what was I even thinking here. 😄

Reply:
I will be returning an empty map and the error and handle the error from where we call this func